### PR TITLE
AddressNotFoundError: add .network

### PR DESCRIPTION
**Pull request by illes**
See `README.rst` snippet for example usage.

### CODE REVIEW

File: geoip2/errors.py [hunk 1]
```diff
@@ -17,7 +19,39 @@ class GeoIP2Error(RuntimeError):
...
+        self.ip_address = ip_address
+        self.prefix_len = prefix_len
+
+    @property
+    def network(self) -> Optional[Union[ipaddress.IPv4Network, ipaddress.IPv6Network]]:
+        """The network for the error"""
+
+        if self.ip_address is None or self.prefix_len is None:
+            return None
+        network = ipaddress.ip_network(f"{self.ip_address}/{self.prefix_len}", False)
```
[the remark this answers is not in the document]
Comment (illes) [reply geoip2/errors.py]: done

File: geoip2/errors.py [hunk 2]
```diff
@@ -17,7 +19,39 @@ class GeoIP2Error(RuntimeError):
...
+
+      :type: str
+
+    .. attribute:: network
+
+      The network associated with the error. In particular, this is the
+      largest network where no address would be found.
+
+    """
+
```
[the remark this answers is not in the document]
Comment (illes) [reply geoip2/errors.py:38]: done

File: geoip2/errors.py [hunk 3]
```diff
@@ -17,7 +19,39 @@ class GeoIP2Error(RuntimeError):
...
+    def __init__(
+        self,
+        message: str,
+        ip_address: Optional[str] = None,
+        prefix_len: Optional[int] = None,
+    ) -> None:
+        super().__init__(message)
+        self.ip_address = ip_address
+        self.prefix_len = prefix_len
```
[the remark this answers is not in the document]
Comment (illes) [reply geoip2/errors.py]: done

File: geoip2/errors.py [hunk 4]
```diff
@@ -17,7 +19,39 @@ class GeoIP2Error(RuntimeError):
...
+
+      The IP address used in the lookup.
+
+      :type: str
+
+    .. attribute:: network
+
+      The network associated with the error. In particular, this is the
+      largest network where no address would be found.
+
```
[the remark this answers is not in the document]
Comment (illes) [reply geoip2/errors.py:34]: done

**Comment by illes** [comment]
@oschwald , added unit test in fixup commit

**Comment by oschwald** [comment]
4.6.0 was released with this change. Thanks!

**Pull request merged**